python/felis/db/_variants.py

```diff
@@ -0,0 +1,90 @@
+# This file is part of felis.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import re
+from typing import Any
+
+from sqlalchemy import types
+from sqlalchemy.dialects import mysql, oracle, postgresql, sqlite
+from sqlalchemy.types import TypeEngine
+
+from ..datamodel import Column
+
+MYSQL = "mysql"
+ORACLE = "oracle"
+POSTGRES = "postgresql"
+SQLITE = "sqlite"
+
+TABLE_OPTS = {
+    "mysql:engine": "mysql_engine",
+    "mysql:charset": "mysql_charset",
+    "oracle:compress": "oracle_compress",
+}
+
+COLUMN_VARIANT_OVERRIDE = {
+    "mysql:datatype": "mysql",
+    "oracle:datatype": "oracle",
+    "postgresql:datatype": "postgresql",
+    "sqlite:datatype": "sqlite",
+}
+
+DIALECT_MODULES = {MYSQL: mysql, ORACLE: oracle, SQLITE: sqlite, POSTGRES: postgresql}
+
+length_regex = re.compile(r"\((.+)\)")
```

Comment:
```suggestion
_length_regex = re.compile(r"\((\d+)\)")
```
This regex could do with an explanation as to what length you are trying to find. It looks like you expect it to be an integer so I'm not sure why it's not `\d+` instead. Underscore since it's a global that you don't want leaking out the file.

Reply:
I inherited this code and copied it from another module I am deleting, but I can look into this. And I will change to have an underscore so it is clear the object is for internal use only.

Reply:
If it really is numbers it's looking for then `\d+` seems better.

Reply:
I updated the regular expression to look for an integer and also prepended the variable name with an underscore.